wordpress-mobile/WordPress-iOS
# Reader: remove unused files

Fixes #n/a
Ref: https://github.com/wordpress-mobile/WordPress-iOS/pull/15382

This removes all (as far as I can tell) unused Reader files. Specifically:

- ReaderSavedPostsViewController
- ReaderSavedPostCellActions
- ReaderMenuViewController
- ReaderFollowedSitesStreamHeader
- ReaderMenuViewModel and related classes.
  - DiscoverMenuItemCreator
  - FollowingMenuItemCreator
  - LikedMenuItemCreator
  - OtherMenuItemCreator
  - ReaderMenuItemCreator
  - SavedForLaterMenuItemCreator
  - SearchMenuItemCreator

There is one more final cleanup step to do in a future PR, and that is to address the `TODO: - READERNAV`s.

To test:

- There should be no functional changes, so just smoke test the Reader and all the filters to ensure they behave as expected. 
- Verify the unit and UI tests pass on this PR.

Note: there is an [existing issue](https://github.com/wordpress-mobile/WordPress-iOS/issues/15381) with unsaving posts from the Saved filter.

PR submission checklist:
